### PR TITLE
Support additional emoji types

### LINKED ISSUE #5: Does not work with zero width joiner emoji
For example 🧑‍🏫 does not add the skin tone variant because adding the skin tone modifier at the end of the emoji is incorrect, it should go before the zero width joiner.

### PR DESCRIPTION
**Pull request by darkmark1991**
fixes #5 + adds support for multi component emojis (couples, detective, etc.)

### Changes

1.  Update the check to see if emoji is skin tone compatible.
- This means that emoji has less than 3 modifiable components (3+ are families, there is no plans to support skin tone for those at this stage https://www.unicode.org/L2/L2020/20114-family-emoji-explor.pdf)
- For the several two-person family emojis additional check was added
2. Exclude `Unicode Presentation Selector` character as it interferes with skin tone application https://unicode.org/reports/tr51/#composing_zwj_seq
3. Apply skin tone to the right of every modifiable character (couples, etc.)
4. Fix skin tone removal to remove from all modifiable characters (couples, etc.)

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for working on this. Can you add some tests?

File: index.js [hunk 1]
```diff
@@ -9,15 +7,36 @@ const skinTones = new Map([
...
+	if (tone === 'none') {
+		return emoji;
+	}
+
+	const baseModifierMatch = (emoji).match(emojiBaseModifierRegex);
+	if (baseModifierMatch !== null && baseModifierMatch.length < 3 && !twoFamilyEmojis.has(emoji)) {
+		const emojiArray = [...emoji];
+		emoji = '';
+
+		for (let i = 0; i < emojiArray.length; i++) {
```
Comment (sindresorhus) [inline index.js]: Use a `for-of` loop. You then also don't need `[...emoji]`.

File: index.js [hunk 2]
```diff
@@ -9,15 +7,36 @@ const skinTones = new Map([
 	['darkBrown', '🏿']
 ]);
 
+const emojiBaseModifierRegex = /\p{Emoji_Modifier_Base}/ug;
+const unicodeVariationSelector = '\uFE0F';
+const twoFamilyEmojis = new Set(['👩‍👦', '👩‍👧', '👨‍👧', '👨‍👦']);
```
Comment (sindresorhus) [inline index.js:18]: This needs a code comment on why it's hard-coded.

File: index.js [hunk 3]
```diff
@@ -9,15 +7,36 @@ const skinTones = new Map([
 	['darkBrown', '🏿']
 ]);
 
+const emojiBaseModifierRegex = /\p{Emoji_Modifier_Base}/ug;
+const unicodeVariationSelector = '\uFE0F';
```
Comment (sindresorhus) [inline index.js]:
```suggestion
const unicodeVariationSelector = '\u{FE0F}';
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by darkmark1991** [comment]
@sindresorhus Hey, thanks for reviewing so quickly! Made requested modifications & added some comments, hope it's better now 🙏

File: index.js [hunk 4]
```diff
@@ -9,15 +7,44 @@ const skinTones = new Map([
 	['darkBrown', '🏿'],
 ]);
 
+// This emoji modifier base \p{Emoji_Modifier_Base} is present in emojis that the skin tone can apply to
+const emojiBaseModifierRegex = /\p{Emoji_Modifier_Base}/ug;
+// Emoji presentation selector takes the same place as skin tone modifier https://unicode.org/reports/tr51/#composing_zwj_seq
+// So it should be removed if present, otherwise it causes issues like this '🕵🏼️‍♀️'
```
Comment (sindresorhus) [inline index.js]: > like this '🕵🏼️‍♀️'

Not immediately clear what the problem with that emoji is. Should be described for clarity.

File: index.js [hunk 5]
```diff
@@ -9,15 +7,44 @@ const skinTones = new Map([
 	['darkBrown', '🏿'],
 ]);
 
+// This emoji modifier base \p{Emoji_Modifier_Base} is present in emojis that the skin tone can apply to
+const emojiBaseModifierRegex = /\p{Emoji_Modifier_Base}/ug;
```
Comment (sindresorhus) [inline index.js]: Regex with global flag is mutable and should not be at the top-level.

File: index.js [hunk 6]
```diff
@@ -9,15 +7,44 @@ const skinTones = new Map([
...
 	if (!skinTones.has(tone)) {
 		throw new TypeError(`Unexpected \`skinTone\` name: ${tone}`);
 	}
 
-	emoji = emoji.replace(/[\u{1F3FB}-\u{1F3FF}]/u, '');
+	emoji = emoji.replaceAll(/[\u{1F3FB}-\u{1F3FF}]/ug, '');
+
+	// If tone is not 'none', emoji has less than 3 modifiable components (i.e. no families)
+	// And is not any of the two-person family emojis, then tone should be applied
+	if (tone !== 'none' && (emoji).match(emojiBaseModifierRegex)?.length < 3 && !twoFamilyEmojis.has(emoji)) {
```
Comment (sindresorhus) [inline index.js]: Do an early return to reduce nesting.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**